### PR TITLE
[MODFEE-72] Rename and extend tokens available to manual fee/fine notices

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
Resolves [MODFEE-72](https://issues.folio.org/browse/MODFEE-72)

### CODE REVIEW

File: src/main/java/org/folio/rest/utils/FeeFineActionCommentsParser.java [hunk 1]
```diff
@@ -1,21 +0,0 @@
-package org.folio.rest.utils;
-
-import java.util.Arrays;
-import java.util.Map;
-import java.util.stream.Collectors;
-
-import org.apache.commons.lang3.tuple.Pair;
-
-public class FeeFineActionCommentsParser {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/utils/FeeFineActionCommentsParser.java:9]: Method moved to `PatronNoticeBuilder`

File: src/test/java/org/folio/PatronNoticeTest.java [hunk 2]
```diff
@@ -1,459 +0,0 @@
...
-import io.vertx.core.Future;
-import io.vertx.core.Handler;
-import io.vertx.core.Promise;
-import io.vertx.core.Vertx;
-import io.vertx.core.json.JsonObject;
-import io.vertx.junit5.VertxExtension;
-import io.vertx.junit5.VertxTestContext;
-
-@ExtendWith(VertxExtension.class)
-class PatronNoticeTest {
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/PatronNoticeTest.java:54]: Logic for choosing the correct `templateId` for the notice is covered with unit tests in `FeeFineNoticeContextTest`. I see no need to test it with API tests, which is done in `FeeFineActionsAPITest` anyway.

File: ramls/inventory/classificationtype.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "A classification type",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
```
Comment (roman-barannyk) [inline ramls/inventory/classificationtype.json:7]: Maybe it makes sense to add description to make it in the same style as other fields

Comment (roman-barannyk) [reply ramls/inventory/classificationtype.json:7]: "$ref": "uuid.json"?

File: ramls/inventory/contributornametype.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "A contributor name type",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
```
Comment (roman-barannyk) [inline ramls/inventory/contributornametype.json:7]: I would add UUID validation and description for this field

File: ramls/inventory/contributornametype.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "ordering": {
+      "description": "used for ordering of contributor name types in displays, i.e. in select lists",
+      "type": "string"
+    },
+    "metadata": {
+      "type": "object",
+      "$ref": "../raml-util/schemas/metadata.schema",
+      "readonly": true
+     }
+  },
```
Comment (roman-barannyk) [inline ramls/inventory/contributornametype.json:22]: What about additional properties? should it be allowed or not?

File: ramls/inventory/illpolicy.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    "metadata": {
+      "type": "object",
+      "$ref": "../raml-util/schemas/metadata.schema",
+      "readonly": true
+    }
+  },
+  "additionalProperties": true,
+  "required": [
+    "name",
+    "source"
```
Comment (roman-barannyk) [inline ramls/inventory/illpolicy.json:27]: You have made "source" required for this schema and not required for others, like  classificationtype.json or contributornametype.json . Is that OK?

File: ramls/inventory/location.json [hunk 7]
```diff
@@ -0,0 +1,133 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "A (shelf) location, the forth-level location unit below institution, campus, and library.",
+  "type": "object",
+  "properties": {
+    "id": {
```
Comment (roman-barannyk) [inline ramls/inventory/location.json:6]: UUID validation?

File: ramls/inventory/location.json [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      "folio:linkToField": "id",
+      "folio:includedElement": "loclibs.0"
+    },
+    "details": {
+      "description": "Details about this (shelf) location.",
+      "type": "object"
+    },
+    "primaryServicePoint": {
+      "description": "The UUID of the primary service point of this (shelf) location.",
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
```
Comment (roman-barannyk) [inline ramls/inventory/location.json:78]: It seems like we already have UUID schema for validation. Why do we need to use this pattern?

File: ramls/inventory/loccamp.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    },
+    "metadata": {
+      "type": "object",
+      "$ref": "../raml-util/schemas/metadata.schema",
+      "readonly": true
+    }
+  },
+  "additionalProperties": true,
+  "required": [
+    "name", "institutionId", "code"
```
Comment (roman-barannyk) [inline ramls/inventory/loccamp.json:29]: maybe it makes sense to to split required fields into multiple lines, one field for one line to be in the same style like in other schemas? or conversely put them on one line for all schemas?

File: ramls/inventory/servicepoint.json [hunk 10]
```diff
@@ -0,0 +1,74 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "description": "A service point",
+  "type": "object",
+  "properties": {
+    "id": {
```
Comment (roman-barannyk) [inline ramls/inventory/servicepoint.json:6]: UUID validation?

File: ramls/loan-policy.json [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+      "type": "string"
+    },
+    "name": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string"
+    },
+    "loanable": {
+      "type": "boolean"
```
Comment (roman-barannyk) [inline ramls/loan-policy.json:15]: description?

File: ramls/loan.json [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+      "description": "Count of how many times a loan has been renewed (incremented by the client)",
+      "type": "integer"
+    },
+    "loanPolicyId": {
+      "description": "ID of last policy used in relation to this loan",
+      "type": "string"
+    },
+    "checkoutServicePointId": {
+      "description": "ID of the Service Point where the last checkout occured",
+      "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$",
```
Comment (roman-barannyk) [inline ramls/loan.json:80]: there is a UUID schema for this checking, I think we don't have to use pattern here

File: ramls/user.json [hunk 13]
```diff
@@ -120,15 +120,15 @@
                 "type": "boolean"
               }
             },
-            "additionalProperties": false
```
Comment (roman-barannyk) [inline ramls/user.json:123]: Do you think it makes sense to allow all additional properties? As I see, only one additional properties is used, it's two levels of json schema hierarchy higher

File: src/main/java/org/folio/rest/client/OkapiClient.java [hunk 14]
```diff
@@ -1,26 +1,40 @@
...
 
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
 import io.vertx.core.buffer.Buffer;
 import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
 import io.vertx.ext.web.client.WebClient;
 
 public class OkapiClient {
+  protected static final Logger log = LoggerFactory.getLogger(OkapiClient.class);
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/client/OkapiClient.java:30]: Why is it protected? Do we use it from somewhere else?

File: src/main/java/org/folio/rest/utils/PatronNoticeBuilder.java [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
+
+    return feeActionContext;
+  }
+
+  private static String getCommentsFromFeeFineAction(Feefineaction feefineaction){
+    String comments = Optional.ofNullable(feefineaction.getComments()).orElse(StringUtils.EMPTY);
+    return parseFeeFineComments(comments).getOrDefault(PATRON_COMMENTS_KEY, StringUtils.EMPTY);
+  }
+
+  static Map<String, String> parseFeeFineComments(String comments) {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/utils/PatronNoticeBuilder.java:256]: It seems like we already have the similar public static method here: org.folio.rest.utils.FeeFineActionCommentsParser#parseFeeFineComments

File: ramls/inventory/servicepoint.json [hunk 16]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (roman-barannyk) [inline ramls/inventory/servicepoint.json:1]: Please check if we need this schema

File: ramls/inventory/materialtype.json [hunk 17]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (roman-barannyk) [inline ramls/inventory/materialtype.json:1]: Please check if we need this schema

File: src/main/java/org/folio/rest/client/InventoryClient.java [hunk 18]
```diff
@@ -32,14 +39,14 @@ public InventoryClient(WebClient webClient, Map<String, String> okapiHeaders) {
...
-      return Future.succeededFuture(new Items()
+      return succeededFuture(new Items()
         .withItems(new ArrayList<>())
         .withTotalRecords(0));
     }
 
     Promise<HttpResponse<Buffer>> promise = Promise.promise();
 
-    HttpRequest<Buffer> request = okapiGetAbs("/inventory/items");
+    HttpRequest<Buffer> request = okapiGetAbs("/item-storage/items");
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/client/InventoryClient.java:49]: Check if it didn't break anything

File: src/main/java/org/folio/rest/client/OkapiClient.java [hunk 19]
```diff
@@ -44,4 +58,46 @@
...
+      if (response.statusCode() != 200) {
+        return failedFuture(format("Failed to get %s by ID. Response status code: %s",
+          objectType.getSimpleName(), response.statusCode()));
+      }
+      try {
+        T object = objectMapper.readValue(response.bodyAsString(), objectType);
+        return succeededFuture(object);
+      } catch (IOException exception) {
+        return failedFuture(format("Failed to parse response for %s. Response body: %s",
+          objectType.getSimpleName(), response.bodyAsString()));
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/client/OkapiClient.java:83]: please add URL logging for failed module

File: src/main/java/org/folio/rest/repository/OwnerRepository.java [hunk 20]
```diff
@@ -12,7 +12,7 @@
 
 public class OwnerRepository {
 
-  private static final String OWNERS_TABLE = "owners";
+  public static final String OWNERS_TABLE = "owners";
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/repository/OwnerRepository.java:15]: please duplicate these constants in test classes

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java [hunk 21]
```diff
@@ -227,36 +304,192 @@ private RequestSpecification getRequestSpecification() {
...
+
+  private static String getNameFromProperties(Map<String, Object> properties) {
+    return (String) properties.get("name");
+  }
+
+  private static String getIdFromProperties(Map<String, Object> properties) {
+    return (String) properties.get("id");
+  }
+
+  static String dateToString(Date date) {
```
Comment (roman-barannyk) [inline src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java:480]: private?

File: ramls/inventory/servicepoint.json (hunk 16 above)
Comment (OleksandrVidinieiev) [reply ramls/inventory/servicepoint.json:1]: yes, we do

File: ramls/inventory/materialtype.json (hunk 17 above)
Comment (OleksandrVidinieiev) [reply ramls/inventory/materialtype.json:1]: yes, we do

File: src/main/java/org/folio/rest/client/OkapiClient.java (hunk 14 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/rest/client/OkapiClient.java]: fixed

File: src/main/java/org/folio/rest/client/InventoryClient.java (hunk 18 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/rest/client/InventoryClient.java:49]: checked

File: src/main/java/org/folio/rest/client/OkapiClient.java (hunk 19 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/rest/client/OkapiClient.java]: done

File: src/main/java/org/folio/rest/repository/OwnerRepository.java (hunk 20 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/rest/repository/OwnerRepository.java]: done

File: src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java (hunk 21 above)
Comment (OleksandrVidinieiev) [reply src/test/java/org/folio/rest/impl/FeeFineActionsAPITest.java]: fixed

**Pull request merged**